stackrox/stackrox
# ROX-13947: Remove Central ownerReference on central-db-password

## Description

the operator creates a garbage-collected central-db-password secret bound to the lifetime of the Central. Though, we don't delete the PVC using garbage collection. When users would reinstall Central, the password would be generated anew, and Central wouldn't be able to connect to the previous database because the password is lost. 

I didn't use the SecretReconciliator because this logic is inherently different from the other generated secrets, who rely on the presence of an owner reference. 

## Checklist

## Testing Performed


  - install the operator 4.0.2 & apply a Central CR
  - wait for the stack to be healthy. Verify that the `central-db-password` has an OwnerReference set to the created Central.
  - delete the Central CR. Wait until everything is uninstalled. 
  - reinstall the Central with the same settings
  - Observe that it cannot connect to the database because the password is wrong
  - Delete the PVC, and the `central-db` Pod, and observe that after a while Central can connect again (restarting the operator makes this quicker)
  - delete the running operator deployment, and start the operator from this branch
  - wait for the reconciliation loop
  - observe that the `central-db-password` does not have an ownerReference anymore
  - verify that central can connect to the database
  - delete the Central CR
  - wait for the resources to be deleted
  - create the Central CR
  - observe that Central can connect to the central-db
